snuba/datasets/entities/discover.py

```diff
@@ -150,6 +154,12 @@ def match_query_to_table(
     elif has_transaction_columns:
         return TRANSACTIONS
     else:
+        # Check for apdex or failure rate
+        for expr in query.get_all_expressions():
+            match = TRANSACTION_FUNCTIONS.match(expr)
+            if match:
+                return TRANSACTIONS
+
```

Comment:
So if I get this correctly this is a functional change with respect to how the entity detector works today. 
Is this inevitable because these processors are not in the discover entity anymore or can it be done as a separate change to reduce the risk (since the rest is iso-functional) ?

Reply:
It needs to be done together because we are moving entity selection from after to before the query processing which is responsible for expanding these functions and adding the references of the transactions specific columns to the query. We don't know that `failure_rate` references the `transaction_status` column at the point of making the decision. On a plus side, we get to completely skip the apdex and failure rate processors unless the transactions entity is picked.